### DISCUSSION TITLE
Extension scripts e.g. manipulating experiments in image viewer

### DISCUSSION THREAD

**Discussion by graeme-winter** [dials/dials#1185] [Ideas]
Thinking inspired by a misunderstanding of @phyy-nx PR - if we could have a way of generating buttons in the image viewer, reciprocal lattice viewer automatically from things found in e.g. `~/.dials/viewer_extensions` with some kind of experiment-modifying-API 

```
script_name = "Thing"

def mod_expt(expt):
    # do things
    return expt
```

then we could have it e.g. rotate the rotation axis about the beam or something which could be useful for experiment / commissioning work?

Picture I have in my head is the directory is searched at run-time and buttons added to run each script which is found in there...

**Comment by phyy-nx** [comment]
Perhaps this is similar to util/image_viewer/slip_viewer/example_frame_plugin_hidden.txt?  We are using plugins for things like calibrating kapton tape shadow correction parameters.

**Comment by phyy-nx** [comment]
Link: https://github.com/dials/dials/blob/master/util/image_viewer/slip_viewer/example_frame_plugin_hidden.txt

[a comment by an automated account is not shown]
